lib/internal/perf/timerify.js

```diff
@@ -80,8 +81,8 @@ function timerify(fn, options = {}) {
     const result = constructor ?
       ReflectConstruct(fn, args, fn) :
       ReflectApply(fn, this, args);
-    if (!constructor && typeof result?.finally === 'function') {
-      return result.finally(
+    if (!constructor && typeof result?.then === 'function') {
+      return result.then(
```

Comment:
What do you mean?

Reply:
Do you mean like this?

https://github.com/nodejs/node/blob/1e761654d3d93e4568d1228b92d1f3a09e92d078/lib/assert.js#L763-L771

Reply:
I mean, if follow JS spec strictly, the code should be:
```js
const {then} = result
if (typeof then === 'function') {
  ReflectApply(then, result, [callback])
}
```
Because theoretically `result.then` could be a getter, and two retrieve could give u different result.

Reply:
I also don't understand why `checkIsPromise` "do not accept thenables that use a function as `obj` and that have no `catch` handler" intentionally, this limitation seems wrong to me.